### PR TITLE
feat: maestro adaptor

### PR DESCRIPTION
**Pull request by SIDANWhatever**
Issues remaining:
1. [X] Pagination - currently i disabled all pagination in the sdk - to be discussed
2. [X] Payment cred - one method in adaptor is unimplemented, need extra clarification on api to proceed
3. [X] `getPoolHistory` - 
    - [X] typescript-sdk needa be [updated](https://github.com/maestro-org/typescript-sdk/pull/37) 
    - [X] existing api misses some elements in current adaptor design in minswap:
```ts
  public async getPoolHistory({
    id,
    count = 100,
    order = "desc",
  }: GetPoolHistoryParams): Promise<PoolHistory[]> {
    const nft = `${POOL_NFT_POLICY_ID}${id}`;
    const nftTxs = await this.api.assets.assetTxs(nft, {
      count,
      order,
    });
    return nftTxs.data.data.map(
      (tx: any): PoolHistory => ({
        txHash: tx.tx_hash,
        txIndex: 0, // To discuss: No such info at response
        blockHeight: 9913642, // To discuss: No such info at response
        time: tx.timestamp,
      })
    );
  }
  ```

### CODE REVIEW

**Comment by SIDANWhatever** [comment]
Blocker - to clarify:
1. [ ] What is / how to obtain the payment cred [this endpoint](https://docs.gomaestro.org/Indexer-API/Addresses/utxos-by-payment-cred) requires:
image

To discuss:
1. [ ] Do we want to perform full pagination feature on this sdk? If so, should we add in our typescript-sdk instead (coz probably everyone needs similar operation)
2. [ ] For `getPoolHistory`, do we need full flags as return type (incl. `txIndex` and `blockHeight`)? From example usage, seems both flags are not used. For full confidence, maybe we need confirmation with minswap team.

File: examples/example.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import {
 import {
   ADA,
   Asset,
-  BlockfrostAdapter,
+  BlockfrostAdaptor,
```
Comment (Vardominator) [inline examples/example.ts:18]: `Adapter` is the correct English spelling. why did you change this?

File: package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
   "homepage": "https://github.com/minswap/sdk#readme",
   "dependencies": {
     "@blockfrost/blockfrost-js": "^5.3.1",
+    "@maestro-org/typescript-sdk": "^1.4.1",
```
Comment (Vardominator) [inline package.json:44]: https://github.com/maestro-org/typescript-sdk/releases/tag/v1.5.0

File: src/index.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export * from "./adapter";
+export * from "./adaptor";
```
Comment (Vardominator) [inline src/index.ts:1]: same as above

File: src/types/adaptor.ts
Comment (Vardominator) [inline src/types/adaptor.ts:1]: Please update filename to `adapter.ts`

File: src/adaptor/maestro-adaptor.ts [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+   */
+  public async getPools({
+    page,
+    count = 100,
+    order = "asc",
+  }: GetPoolsParams): Promise<PoolState[]> {
+    const paymentCred = getPaymentCredFromScriptHash(POOL_SCRIPT_HASH);
+    // console.log(page);
+
+    // const res = await this.api.addresses.utxosByPaymentCred(paymentCred, {
```
Comment (Vardominator) [inline src/adaptor/maestro-adaptor.ts:50]: remove debug comments

**Review by Vardominator — APPROVED** [review]
